src/main/java/io/github/moulberry/notenoughupdates/profileviewer/ProfileViewerUtils.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright (C) 2023 NotEnoughUpdates contributors
+ *
+ * This file is part of NotEnoughUpdates.
+ *
+ * NotEnoughUpdates is free software: you can redistribute it
+ * and/or modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation, either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * NotEnoughUpdates is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with NotEnoughUpdates. If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package io.github.moulberry.notenoughupdates.profileviewer;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonPrimitive;
+import io.github.moulberry.notenoughupdates.NEUManager;
+import io.github.moulberry.notenoughupdates.NotEnoughUpdates;
+import io.github.moulberry.notenoughupdates.util.Constants;
+import io.github.moulberry.notenoughupdates.util.JsonUtils;
+import io.github.moulberry.notenoughupdates.util.Utils;
+import net.minecraft.nbt.CompressedStreamTools;
+import net.minecraft.nbt.JsonToNBT;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.nbt.NBTTagList;
+import net.minecraft.util.EnumChatFormatting;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.util.AbstractMap;
+import java.util.Base64;
+import java.util.Comparator;
+import java.util.HashSet;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class ProfileViewerUtils {
+
+	public static JsonObject readInventoryInfo(JsonObject profileInfo, String bagName) {
+		JsonObject inventoryInfo = new JsonObject();
+		JsonElement element = Utils.getElement(profileInfo, bagName + ".data");
+
+		String bytes = Utils.getElementAsString(element, "Hz8IAAAAAAAAAD9iYD9kYD9kAAMAPwI/Gw0AAAA=");
```

Comment:
This string does not read as proper gzip or proper nbt data to me? Maybe im doing something wrong:

```
λ echo Hz8IAAAAAAAAAD9iYD9kYD9kAAMAPwI/Gw0AAAA= | base64 -d | gunzip - 
gzip: stdin: not in gzip format
```

Reply:
Compare it with ProfileViewer 1464
I have not changed the logic on that?

Reply:
![image](https://user-images.githubusercontent.com/20768569/216827637-fe45dfa9-6f16-4f9e-9f4f-0c3c0e6a5bca.png)
This is neither valid GZIP Nbt data, nor valid uncompressed Nbt data.

Reply:
I just copied Moulberry code from the Profile Viewer class. IDK for what the `Hz8IAAAAAAAAAD9iYD9kYD9kAAMAPwI/Gw0AAAA=` text stands for at all. It just works for me. If you think you know what it's meant to represent and know how to fix it (if there is anything we should fix at all since it's working for me) then please tell me.

Reply:
That string is the base64 encoded version of a gzipped nbt tag that is representing an empty inventory. This is a fallback for when the api does not provide the talisman bag inventory (which would be in the same format)

Reply:
So its fine?

Reply:
No, thats what the string is supposed to be, but its just a wrong string that does not decode at all.

Reply:
So, we should just remove the fallback case in the context of this PR since we don't want to represent an empty inventory and leave the wrong string in the profile viewer class?

Reply:
i mean in either case i dont like the idea of using a string for the fallback. i much prefer just an `if(x== null) return new ArrayList<ItemStack>()` (completely bypassing the nbt decoding step) than having some weird string that nobody understands. if you wanna fix that is up to you (since the exception caused by this invalid string is caught anyway, so its not a crash, just bad code)